### ISSUE TITLE
Low hashrate on GTX 1070

### ISSUE THREAD

**Issue by PaulieLUL** [fireice-uk/xmr-stak-nvidia#209]
Hello, my miner is crashing using single EVGA GTX 1070 FTW with the default nvidia.txt settings : 
"threads" : 64, "blocks" : 45,
    "bfactor" : 6, "bsleep" :  25,
I am getting following error: 
![screenshot_2](https://user-images.githubusercontent.com/34744253/34258358-e0af3364-e65d-11e7-9eb9-628ad39faba6.png)

I tried to play with the settings and I ended up with:
"threads" : 32, "blocks" : 15,
    "bfactor" : 8, "bsleep" :  25,
which gives me about 530 H/s even tho my GPU should do 700 H/s no problem

____________________________________________________
I am using default nvidia.txt settings on my second PC with GTX 960 and the miner isn't crashing and everything works fine there.

**Comment by marwi1** [comment]
For GTX 1070:
 "gpu_threads_conf" : 
  [
      { "index" : 0, "threads" : 64, "blocks" :45, "bfactor" : 6, "bsleep" :  25, "affine_to_cpu" : false},
  ],
about  769 H/s

**Comment by Fomych** [comment]
I have crashes by  "threads" : 64, "blocks" :45.  What should i do to have a good H/s ?

**Comment by lug-gh** [comment]
"threads": 15,
"blocks": 30,
"bfactor": 6,
"bsleep": 25,
"affine_to_cpu": false

2050/4400 Mhz = 680 H/s

**Comment by Fomych** [comment]
i have by
"threads": 65,
"blocks": 40,
610 H/s
and by 
"threads": 15,
"blocks": 30,  under 550 H/s.

**Comment by Fomych** [comment]
i have by
"threads": 65,
"blocks": 40,
610 H/s
and by
"threads": 15,
"blocks": 30, under 550 H/s.

**Comment by lug-gh** [comment]
what are your cards core and mem clock?

change blocks to 60, it made my card do 710 H/s

bfactory 8 bsleep 100

**Comment by Fomych** [comment]
Core Clock :   Boost: 1822 MHz/ Base: 1620 MHz in OC Mode 
                      Boost: 1784 MHz/ Base: 1594 MHz in Gaming Mode
Memory Clock:	8008 MHz

And if i change to more than 40 blocks it crashing

**Comment by lug-gh** [comment]
Do you have Afterburner or any other tools to overlock your card?

Mine runs at 2050Mhz Core Clock @ 931mV (undervolted) and 4404MHz Memory (or 8808MHz as you'd say) -> http://www.overclock.net/forum/74-graphics-cards-general/320291-why-does-gpu-memory-clock-show-up-1-2-a.html

Overclocking the RAM ist a must for cryptonight algorhythm.

**Comment by grtbaldini** [comment]
I had a windows update happened and it killed my rig. I had let it update the important things and thought I was good. It updated that Fall creators update.  Rig running since Dec, Then this past Monday boom! So my hashrates on my 1070's are hitting like 9.0 mh. I can't figure it out after I had to reformat and install 10 again. Sorry to hijack your thread. It came up in my search

**Comment by marwi1** [comment]
Card Palit GTX 1070 DUAL. **1898/4657 MHz** 
Windows10 
**set bfactor=7,
set launchParam=--launch=75x45,
set bSleep=15**
![palithash](https://user-images.githubusercontent.com/34866291/35716635-73b850a8-07da-11e8-98f6-3efd69e6c480.jpg)
The card charges approx. 100W
Temperature about 50 degrees Celsius
![mzad](https://user-images.githubusercontent.com/34866291/35717107-68ab6288-07dd-11e8-91ca-07fbf3671496.jpg)

**Comment by lug-gh** [comment]
How can you Hit 4500MHz+ without getting fragments/picture errors on the screen in windows?

> set bfactor=7,
> set launchParam=--launch=75x45,
> set bSleep=15

gives me 720 H/s at 2050/4404

**Comment by Fomych** [comment]
With this setings 
set bfactor=7,
set launchParam=--launch=75x45,
set bSleep=15
I have 611 H/s.
I will try to boost my card, maybe it will help

**Comment by lug-gh** [comment]
Don't use 75x45, use 15x30 or 15x60, as the 1070's SMC limit is 15. Choosing 75 will not improve performance, it will just eat more ram.

**Comment by marwi1** [comment]
You're right that 75x45 uses the entire memory of the card. With the setting of 15x60, the performance drops to 645 H / s.

**Comment by Fomych** [comment]
15x60 610 H/s =(
